app/routes/cdo.js

```diff
@@ -96,5 +98,50 @@ module.exports = [
         throw e
       }
     }
+  },
+  {
+    method: 'POST',
+    path: '/cdo/{indexNumber}/manage:recordInsuranceDetails',
+    options: {
+      validate: {
+        payload: recordInsuranceDetailsSchema,
+        failAction: (request, h, err) => {
+          console.error(err)
+
+          return h.response({ errors: err.details.map(e => e.message) }).code(400).takeover()
+        }
+      },
+      response: {
+        schema: recordInsuranceDetailsResponseSchema,
+        failAction: (request, h, err) => {
+          console.error(err)
+
+          return h.response({ errors: err.details.map(e => e.message) }).code(400).takeover()
+        }
+      },
+      handler: async (request, h) => {
+        const indexNumber = request.params.indexNumber
+        const { insuranceCompany, insuranceRenewal } = request.payload
```

Comment:
A tiny thing here but pulling out these two vars from the payload, then passing them both back in to recordInsuranceDetails() seems unnecessary. You could just pass the payload in (assuming the payload is validated to only contain those two elements)